apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/SolrClientInterceptor.java

```diff
@@ -0,0 +1,194 @@
+                    span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+
+                    span.tag(SolrjTags.TAG_COMMIT, "true");
+                    span.tag(SolrjTags.TAG_SOFT_COMMIT, params.get(UpdateParams.SOFT_COMMIT, ""));
+                } else {
+                    span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+
+                    span.tag(SolrjTags.TAG_OPTIMIZE, "true");
+                    span.tag(SolrjTags.TAG_MAX_OPTIMIZE_SEGMENTS, params.get(UpdateParams.MAX_OPTIMIZE_SEGMENTS, "1"));
+                }
+            }
+            span.tag(SolrjTags.TAG_ACTION, actionName);
+        } else if (request instanceof QueryRequest) { // It failed to spot that through 'path' when it had defined QueryType.
+            String operatorName = String.format("solrJ/%s%s", collection, request.getPath());
+            span = ContextManager.createExitSpan(operatorName, instance.getRemotePeer())
+                    .setComponent(ComponentsDefine.SOLRJ)
+                    .setLayer(SpanLayer.DB);
+        } else { // No admin API
+            return;
+        }
+
+        span.tag(SolrjTags.TAG_PATH, request.getPath());
+        span.tag(SolrjTags.TAG_COLLECTION, collection);
+        span.tag(SolrjTags.TAG_METHOD, request.getMethod().name());
+        span.tag(SolrjTags.TAG_QT, params.get(CommonParams.QT, request.getPath()));
+
+        Context.get().start();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                              Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        if (!ContextManager.isActive()) return ret;
+
+        final Context context = Context.remove();
+        final long elapse = System.currentTimeMillis() - context.getStartTime();
+
+        AbstractSpan span = ContextManager.activeSpan();
+        NamedList<Object> response = (NamedList<Object>) ret;
+        if (response != null) {
+            NamedList<Object> header = (NamedList<Object>) response.get("responseHeader");
+            if (header != null) { // common
+                span.tag(SolrjTags.TAG_STATUS, header.get("status").toString());
+                span.tag(SolrjTags.TAG_Q_TIME, header.get("QTime").toString());
+            }
+            SolrDocumentList list = (SolrDocumentList) response.get("response");
+            if (list != null) { // query
+                span.tag(SolrjTags.TAG_NUM_FOUND, String.valueOf(list.getNumFound()));
+            }
+        }
+        SolrjTags.addHttpResponse(span, context);
+        SolrjTags.addElapseTime(span, elapse);
+
+        ContextManager.stopSpan();
+        return ret;
+    }
+
+    @Override
+    public void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Throwable t) {
+        ContextManager.activeSpan().errorOccurred().log(t);
```

Comment:
span is maybe not the solr span

Reply:
Suggest adding`if (!ContextManager.isActive()) return ret;` to avoid NPE.